src/main/java/org/reficio/p2/P2Mojo.java

```diff
@@ -545,14 +549,18 @@ private void executeCategoryPublisher() throws AbstractMojoExecutionException, I
         publisher.execute();
     }
 
+    @SuppressFBWarnings("DM_DEFAULT_ENCODING")
     private void prepareCategoryLocationFile() throws IOException {
         if (categoryFileURL == null || categoryFileURL.trim().isEmpty()) {
-            InputStream is = getClass().getResourceAsStream(DEFAULT_CATEGORY_CLASSPATH_LOCATION + DEFAULT_CATEGORY_FILE);
-            File destinationFolder = new File(destinationDirectory);
-            destinationFolder.mkdirs();
-            File categoryDefinitionFile = new File(destinationFolder, DEFAULT_CATEGORY_FILE);
-            FileWriter writer = new FileWriter(categoryDefinitionFile);
-            IOUtils.copy(is, writer, "UTF-8");
+            File categoryDefinitionFile;
+            FileWriter writer;
+            try (InputStream is = getClass().getResourceAsStream(DEFAULT_CATEGORY_CLASSPATH_LOCATION + DEFAULT_CATEGORY_FILE)) {
+                File destinationFolder = new File(destinationDirectory);
+                destinationFolder.mkdirs();
+                categoryDefinitionFile = new File(destinationFolder, DEFAULT_CATEGORY_FILE);
+                writer = new FileWriter(categoryDefinitionFile);
```

Comment:
… use UTF-8 here, because the next line uses UTF-8 anyway.

Reply:
… this comment should have highlighted the `new FileWriter()` call.

Reply:
> … this comment should have highlighted the `new FileWriter()` call.

Yes that true. But Filewriter constructor has no constructor with encoding. so I have to replace it by `new OutputStreamWriter(new FileOutputStream(categoryDefinitionFile), StandardCharsets.UTF_8);